drivers/include/soft_spi.h

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    driver_soft_spi Soft SPI
+ * @ingroup     drivers
+ * @brief       Software implemented Serial Peripheral Interface Bus
+ * @{
+ *
+ * @file
+ * @brief       Software SPI port descriptor definition
+ *
+ * @author      Markus Blechschmidt <Markus.Blechschmidt@haw-hamburg.de>
+ */
+
+#ifndef SOFT_SOFT_SPI_H
+#define SOFT_SOFT_SPI_H
+
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "mutex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Default SPI device access macro
+ */
+#ifndef SOFT_SPI_DEV
+#define SOFT_SPI_DEV(x)      (x)
+#endif
+
+/**
+ * @brief   Define global value for undefined SPI device
+ */
+#ifndef SOFT_SPI_UNDEF
+#define SOFT_SPI_UNDEF       (UINT_MAX)
+#endif
+
+/**
+ * @brief   Define value for unused CS line
+ */
+#ifndef SOFT_SPI_CS_UNDEF
+#define SOFT_SPI_CS_UNDEF    (GPIO_UNDEF)
+#endif
+
+/**
+ * @brief   Default type for SPI devices
+ */
+typedef unsigned int soft_spi_t;
+
+/**
+ * @brief   Chip select pin type overlaps with gpio_t so it can be casted to
+ *          this
+ */
+#ifndef HAVE_SOFT_SPI_CS_T
+typedef gpio_t soft_spi_cs_t;
+#endif
+
+/**
+ * @brief   Status codes used by the SPI driver interface
+ */
+enum {
+    SOFT_SPI_OK          =  0,   /**< everything went as planned */
+    SOFT_SPI_NODEV       = -1,   /**< invalid SPI bus specified */
+    SOFT_SPI_NOCS        = -2,   /**< invalid chip select line specified */
+    SOFT_SPI_NOMODE      = -3,   /**< selected mode is not supported */
+    SOFT_SPI_NOCLK       = -4    /**< selected clock value is not supported */
+};
+
+/**
+ * @brief   Available SPI modes, defining the configuration of clock polarity
+ *          and clock phase
+ *
+ * RIOT is using the mode numbers as commonly defined by most vendors
+ * (https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus#Mode_numbers):
+ *
+ * - MODE_0: CPOL=0, CPHA=0 - The first data bit is sampled by the receiver on
+ *           the first SCK rising SCK edge (this mode is used most often).
+ * - MODE_1: CPOL=0, CPHA=1 - The first data bit is sampled by the receiver on
+ *           the second rising SCK edge.
+ * - MODE_2: CPOL=1, CPHA=0 - The first data bit is sampled by the receiver on
+ *           the first falling SCK edge.
+ * - MODE_3: CPOL=1, CPHA=1 - The first data bit is sampled by the receiver on
+ *           the second falling SCK edge.
+ */
+#ifndef HAVE_SOFT_SPI_MODE_T
```

Comment:
Ok what is the intention behind these checks? Is it just copy paste from spi.h?

Reply:
jup, copy paste error
:heavy_check_mark: